cpp/src/io/csv/csv_reader.cu

```diff
@@ -552,7 +554,7 @@ gdf_error read_csv(csv_read_arg *args)
 	}
 	else { checkError(GDF_C_ERROR, "invalid input type"); }
 
-	const char* h_uncomp_data;
+	const char* h_uncomp_data = nullptr;
```

Comment:
For empty files that have an extension for one of the compressed formats, like `.gzip`, the call to `getUncompressedHostData()` will fail.

Would it be better to still return an empty dataframe for those cases?

Reply:
Modified the code to completely skip the decompression step if the input is empty. The step makes no sense to execute, as map_data (input for this step) is null in this case.